src/store.ts

```
@@ -0,0 +1,369 @@
+          set({
+            isLoggedIn: loggedIn,
+            ...(!loggedIn ? { userInfoCache: undefined } : {}),
+          });
+        },
+        fetchJsonWithAuth<T>(url: string): Promise<T> {
+          const { baseUrl, csrfToken } = get();
+
+          return fetch(baseUrl + url, {
+            headers: {
+              Authorization: 'Bearer ' + csrfToken,
+            },
+            credentials: 'include',
+          }).then<T>((response) => {
+            if (!response.ok) {
+              throw new HttpError({ statusCode: response.status, message: 'Error fetching JSON' });
+            }
+            return response.json();
+          });
+        },
+
+        authorize(params: Params = {}) {
+          const { defaultAuthConfig, baseUrl } = get();
+          const queryParams = { ...defaultAuthConfig, ...params };
+          if (!queryParams.redirect_uri) {
+            queryParams.redirect_uri = stripTokenFromUrl(window.location.href);
+          }
+          window.location.href = baseUrl + '/authorize?' + buildQuerystring(queryParams);
+        },
+
+        logout: (params: Params = {}) => {
+          const { baseUrl } = get();
+          const queryParams = {
+            ...params,
+            post_logout_redirect_uri: params.post_logout_redirect_uri || window.location.href,
+          };
+
+          localStorage.removeItem(LOGGED_IN_TOKEN_STORAGE_KEY);
+          localStorage.removeItem(USER_INFO_TOKEN_STORAGE_KEY);
+          window.location.href = baseUrl + '/logout?' + buildQuerystring(queryParams);
+        },
+
+        receiveSessionToken(redirect = true) {
+          const { methods: { setSessionToken, getUserInfo } } = get();
+          const match = window.location.search.match(/[?&]token=([^&]+)/) || [];
+          if (!match?.length) return;
+
+          setSessionToken(match[1]);
+          getUserInfo()
+            .finally(() => {
+              if (redirect) {
```

fail fast
if(!redirect) return

---

als geen redirect is moet hij als nog sessie controleren of userinfo update dus dat gaat niet op denk ik?